Sprint 05

KosAleks opened the pull request.

KosAleks committed:
Created new counter for record. Fixed total accuracy.

(cherry picked from commit ced6d8cdb719d4e65519c28f03f440753c11d6eb)

KosAleks committed:
Formatted code sprint_05.

KosAleks committed:
Final version sprint_05

So1ovyov commented on MovieQuiz/Models/BestGameModel.swift:
```diff
@@ -6,23 +6,17 @@
-    
-   
+}
+
```
Лишние пустые строки можно удалить, чтобы визуально не увеличивать файл.

So1ovyov commented on MovieQuiz/Services/Presentation/AlertPresenter.swift:
```diff
@@ -25,41 +22,18 @@ final class AlertPresenter: AlertPresenterProtocol {
+        
+        alertDelegate?.present(alert, animated: true)
+        alert.addAction(action)
+        // self.present(alert, animated: true, completion: nil) // present - предоставить. Эта функция автоматом заложена в UIAlertAction
```
Закомментированный код нужно удалить, если он не используется

So1ovyov commented on MovieQuiz/Services/Presentation/MovieQuizViewController.swift:
```diff
@@ -16,58 +16,19 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
+    private var questionFactory: QuestionFactoryProtocol?
+    private var currentQuestion: QuizQuestion?
     private var alertPresenter: AlertPresenterProtocol?
     private var statisticService: StatisticServiceImplementation?
```
Данная переменная должна быть типа `StatisticService` не имплементация

So1ovyov commented on MovieQuiz/Services/Presentation/MovieQuizViewController.swift:
```diff
@@ -16,58 +16,19 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
-            image: UIImage(data: model.image) ?? UIImage(),
-            question: model.text,
-            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)")
+        let questionStep = QuizStepViewModel(image: UIImage(named: model.image) ?? UIImage(), question: model.text, questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)")
```
Давай длинную строку разделять, чтобы визуально легче читалось, например:
```
        let questionStep = QuizStepViewModel(
            image: UIImage(named: model.image) ?? UIImage(),
            question: model.text,
            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)"
        )
```

So1ovyov commented on MovieQuiz/Services/Presentation/MovieQuizViewController.swift:
```diff
@@ -103,40 +64,40 @@ final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
+            assertionFailure("error")
+            return ""}
+        
+        let string1 = "Ваш результат: \(correctAnswers) / \(questionsAmount)"
```
Давай сделаем нейминг более предметным, чтобы было понятно где какая строка в коде:
```
        let resultGame = "Ваш результат: \(correctAnswers) / \(questionsAmount)"
        let countGame = "Количество сыграных квизов: \(statisticService.gamesCount)"
        let recordGame = "Рекорд: \(bestGame.correct) / \(bestGame.total) \n \(bestGame.date.dateTimeString)"
        let totalAccuracy = "Средняя точность: \(String(format:"%.2f", statisticService.totalAccuracy)) %"
```

So1ovyov commented on MovieQuiz/Services/StatisticServiceImplementation.swift:
```diff
@@ -66,17 +66,17 @@ final class StatisticServiceImplementation: StatisticServiceProtocol {
+        
     }
     
     // MARC -- Metods
```
Корректно будет:
```
// MARK: - Methods
```

KosAleks committed:
Fixed Allert: design according to figma layout.
StatisticServiceImplementation closed by protocol StatisticServiceProtocol.
MovieQuizViewController have private var statisticService type StatisticServiceProtocol.
